### PR TITLE
Fix routes, redirects, and basic auth

### PR DESCRIPTION
**Pull request by jameswilson**
* Fixes regex to detect Environment configuration optional colon (:)  which it seems they may have modified the output string for Environment Config info since the original regular expression was created.

* Adds support for `environment.activate` webhook. 
[platformsh2slack.environment.activate.1537830200.json.txt](https://github.com/hanoii/platformsh2slack/files/2414226/platformsh2slack.environment.activate.1537830200.json.txt)
screen shot 2018-09-25 at 2 16 54 am

* Adds support for `environment.update.http_access` webhook.
[platformsh2slack.environment.update.http_access.1537830227.json.txt](https://github.com/hanoii/platformsh2slack/files/2414227/platformsh2slack.environment.update.http_access.1537830227.json.txt)

screen shot 2018-09-25 at 2 26 51 am

* Adds support for `environment.access.add`, `environment.access.remove`,  `environment.redeploy` and `environment.syncrhonize`. (We're finding these particularly useful for sharing the HTTP Auth info in Slack which we use extensively on all non-master environments).

* Improves display of Routes and Redirects based on payload data instead of preg_match.

* Adds option to exclude redirects from some messages. Note: environment.activate and environment.branch webhooks will always show redirects.

### CODE REVIEW

**Comment by hanoii** [comment]
Hey @jameswilson , thanks for this. I was meaning to get to it, and pretty much accept it, but wanted to give it a proper look. Will try to do it soon.

**Comment by jameswilson** [comment]
Hi @hanoii just a friendly ping.  We're definitely using these changes successfully in our teams at @BluesparkLabs. We love the ability to see the HTTP Auth strings right there in Slack.

File: src/Platformsh2Slack.php [hunk 1]
```diff
@@ -251,14 +305,88 @@ function processPlatformshPayload() {
...
+            'title' => 'HTTP Authentication',
+            'text' => '',
+            'fallback' => $basic_auth_fallback,
+            'fields' => $basic_auth_fields,
+            'color' => $this->config['attachment_color'],
+          ));
+        }
+      }
+
+      if ($platformsh->result != 'success') {
```
Comment (hanoii) [inline src/Platformsh2Slack.php:382]: Any reason why you are doing this just for the basic auth or this was meant to happen to every commit?

**Comment by hanoii** [comment]
@jameswilson I merged it anyway, I can come up to any changes later, but see if you can check on my comment above in the code.

File: src/Platformsh2Slack.php (hunk 1 above)
Comment (jameswilson) [reply src/Platformsh2Slack.php:382]: Hi @hanoii, happy new year. I'm assuming your reference to "doing this" is referring to the conditional section that starts with `if ($platformsh->result != 'success') {`.  I should point out that lines 349 through 382 have to do with the basic auth, however the 'success' conditional comes after that, so it is not scoped to basic auth, but to every platform command interaction.  Basically:  if the command fails, show the log message.  Make sense?

Comment (jameswilson) [reply src/Platformsh2Slack.php:382]: whoops, that should have read "lines 349 through 380", not 382.

Comment (jameswilson) [reply src/Platformsh2Slack.php:382]: Also,  Thanks for the merge!

**Pull request merged**